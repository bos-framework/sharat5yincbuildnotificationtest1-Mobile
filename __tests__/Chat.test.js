import React from 'react';
import Chat from '../src/Screens/Container/Modules/Chat';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Render Chat component without crashing', () => {

  it('Render Chat component without crashing', () => {
    const tree = renderer
      .create(<Chat />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Pass data to state of component.
  it('componentWillMount: set state values', () => {
    const wrapper = shallow(<Chat />);
    wrapper.setState({ subModules: ["Teacher", "parent"] })
    expect(wrapper.state('subModules')).toEqual(["Teacher", "parent"]);
  });

  // Find the controls/elements/child component in the Chat component.
  it('Find components/Elements', () => {

    const wrapper = shallow(<Chat />);
    expect(wrapper.find('View').length).toEqual(3);
  });
});