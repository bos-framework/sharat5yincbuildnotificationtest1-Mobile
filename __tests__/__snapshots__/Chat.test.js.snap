// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Chat component without crashing Render Chat component without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "marginTop": 10,
        }
      }
    >
       Module -&gt; Sub Modules 
    </Text>
    <Text>
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 0.5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
       Module -&gt; Operations 
    </Text>
    <Text>
      
    </Text>
  </View>
</View>
`;
